frontend/src/diagram/sprotty/utils/geometry.tsx

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2022 THALES GLOBAL SERVICES.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { Bounds, Point } from 'sprotty';
+
+export enum RectangleSide {
+  'north',
+  'south',
+  'west',
+  'east',
+}
+/**
+ * Returns the distance between the point p and the segment made of p1 and p2
+ * @param p the point
+ * @param p1 the first point of the segment
+ * @param p2 the second point of the segment
+ */
+export function closestPointOnSegment(p: Point, p1: Point, p2: Point): Point {
+  var A = p.x - p1.x;
+  var B = p.y - p1.y;
+  var C = p2.x - p1.x;
+  var D = p2.y - p1.y;
```

Comment:
Could you have more meaningful variable names than literally A, B, C and D? and not in upper case and not with the `var` keyword. Use either `const` or `let` if they should be modified.